test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeAgent.cs

```diff
@@ -1,36 +1,27 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace Microsoft.Azure.Devices.Edge.Test.Common
 {
-    using System;
     using System.Threading;
     using System.Threading.Tasks;
-    using Microsoft.Azure.Devices.Edge.Util;
 
     public class EdgeAgent : EdgeModule
     {
+        IotHub iotHub;
+
         public EdgeAgent(string deviceId, IotHub iotHub)
-            : base("edgeAgent", deviceId, iotHub)
+            : base("edgeAgent", deviceId)
         {
+            this.iotHub = iotHub;
         }
 
-        public Task PingAsync(CancellationToken token)
-        {
-            return Profiler.Run(
-                () =>
-                {
-                    return Retry.Do(
-                        () => this.iotHub.InvokeMethodAsync(
-                            this.deviceId,
-                            $"${this.Id}",
-                            new CloudToDeviceMethod("ping"),
-                            token),
-                        result => result.Status == 200,
-                        e => true,
-                        TimeSpan.FromSeconds(5),
-                        token);
-                },
+        public Task PingAsync(CancellationToken token) =>
+            Profiler.Run(
+                () => this.iotHub.InvokeMethodAsync(
+                    this.deviceId,
+                    $"${this.Id}",
+                    new CloudToDeviceMethod("ping"),
```

Comment:
For learning, do we have "ping" direct method handling in Edge Agent or just implement for testing?  And what will it respond?

Reply:
It's implemented in production, see edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/PingRequestHandler.cs. It does nothing, just responds with 200 OK. The portal used to use it to show edge connection status, although I don't think they do it that way anymore.